src/responses/search-response-details.ts

```diff
@@ -77,16 +83,38 @@ export class SearchResponseDetails {
     this.schema = schema;
     const schemaHitType = schema?.hit_type;
 
-    this.totalResults = body?.hits?.total ?? 0;
-    this.returnedCount = body?.hits?.returned ?? 0;
-    this.results =
-      body?.hits?.hits?.map((hit: SearchResult) =>
-        SearchResponseDetails.createResult(hit.hit_type ?? schemaHitType, hit)
-      ) ?? [];
+    let firstPageElement;
+    if (body?.page_elements) {
+      this.pageElements = body.page_elements;
+      if (this.pageElements) firstPageElement = Object.values(this.pageElements)[0];
+    }
+
+    // Use hits directly from the body if available.
+    // Otherwise, try extracting them from the first page_element
+    let hits = body?.hits?.hits;
+    if (!hits?.length && this.pageElements) {
+      if (firstPageElement?.hits?.hits) {
+        hits = firstPageElement.hits.hits;
+      }
+    }
+
+    // Use aggregations directly from the body if available.
+    // Otherwise, try extracting them from the first page_element.
+    let aggregations = body?.aggregations;
+    const bodyHasAggregations = this.aggregations && Object.keys(this.aggregations).length > 0;
+    if (!bodyHasAggregations && this.pageElements && firstPageElement?.aggregations) {
+      aggregations = firstPageElement.aggregations;
+    }
+
+    this.totalResults = body?.hits?.total ?? firstPageElement?.hits?.total ?? 0;
+    this.returnedCount = body?.hits?.returned ?? firstPageElement?.hits?.returned ?? 0;
```

Comment:
q: should we throw an error here to be caught - `0` is a good default value, but it can be a red herring if service breaks..

Reply:
Good point! One catch here is that not every response from the PPS necessarily has these total/returned fields, so it isn't always an error for them not to be present. For instance, a request for the forum posts tab does not contain these fields.

Perhaps we can condition this on the type of response that was received, and only throw an error for response types that we expect to have these properties.